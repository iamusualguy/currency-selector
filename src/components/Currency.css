.currency {
  color: #124572;
  background: #ececec;
  border: 1px solid #a3a3a3;
  width: 9em;
  height: 2em;
  line-height: 2;
  margin: 0.5em 0.3em;
  cursor: pointer;
  user-select: none;
}

.currency:not(.selected):hover {
  background: #ddd;
}

.currency.selected {
  background: #fff;
}

input[type=checkbox] {
  -moz-appearance: none;
  -webkit-appearance: none;
  -o-appearance: none;
  outline: none;
  content: none;
  pointer-events: none;
}

input[type=checkbox]:before {
  content: "X";
  font-size: 1em;
  color: transparent !important;
  background: #0000;
  display: block;
  width: 0.9em;
  height: 0.9em;
  border-radius: 0.1em;
  border: 0.05em solid #a3a3a3;
  font-weight: bold;
  line-height: 1;
  text-align: center;
}

input[type=checkbox]:checked:before {
  color: red !important;
  background: #fff;
}