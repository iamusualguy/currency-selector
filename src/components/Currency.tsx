import React from 'react';
import './Currency.css';

export interface CurrencyProps {
  currency: string;
  isSelected: boolean;
  onClick: () => void;
}

export const Currency: React.FC<CurrencyProps> = (props) => {
  return (
    <div
      className={`currency ${props.isSelected ? "selected" : ""}`}
      onClick={() => props.onClick()}
    >
      <input
        type="checkbox"
        defaultChecked={props.isSelected}
        data-testid="currency-checkbox"
      />
      {props.currency}
    </div>
  );
}
