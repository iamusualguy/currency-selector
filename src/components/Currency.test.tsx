import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Currency } from './Currency';

describe("Currency", () => {
  const noop = () => { };

  it('renders Currency component', () => {
    const { asFragment } = render(<Currency currency="USD" isSelected={false} onClick={noop} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders selected Currency component', () => {
    const { container, getByTestId } = render(<Currency currency="USD" isSelected={true} onClick={noop} />);
    const checkbox = getByTestId("currency-checkbox") as HTMLInputElement;
    expect(checkbox.checked).toEqual(true);
    expect(container.firstChild).toHaveClass("selected");
  });
});