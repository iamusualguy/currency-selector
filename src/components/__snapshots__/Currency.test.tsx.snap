// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency renders Currency component 1`] = `
<DocumentFragment>
  <div
    class="currency "
  >
    <input
      data-testid="currency-checkbox"
      type="checkbox"
    />
    USD
  </div>
</DocumentFragment>
`;
